thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/onboard/tasks/FunctionCreationOnboardingTask.java

```diff
@@ -105,40 +117,66 @@ public void run() {
     Preconditions.checkNotNull(configuration.getString(FUNCTION_NAME));
     Preconditions.checkNotNull(configuration.getString(COLLECTION_NAME));
     Preconditions.checkNotNull(configuration.getString(METRIC_NAME));
-    Preconditions.checkNotNull(configuration.getString(CRON_EXPRESSION));
-    Preconditions.checkNotNull(configuration.getString(WINDOW_SIZE));
-    Preconditions.checkNotNull(configuration.getString(WINDOW_UNIT));
-    Preconditions.checkNotNull(configuration.getString(PROPERTIES));
     Preconditions.checkArgument(configuration.containsKey(ALERT_ID) || configuration.containsKey(ALERT_NAME));
     if (!configuration.containsKey(ALERT_ID)) {
       Preconditions.checkNotNull(configuration.getString(ALERT_TO));
     }
 
+    // check if duplicate name exists
+    List<AnomalyFunctionDTO> duplicateFunctions = anoomalyFunctionDAO.findWhereNameLike(configuration.getString(FUNCTION_NAME));
+    for (AnomalyFunctionDTO duplicateFunction : duplicateFunctions) {
+      if (duplicateFunction.getFunctionName().equals(configuration.getString(FUNCTION_NAME))) {
+        throw new IllegalArgumentException("Duplicate function name " + configuration.getString(FUNCTION_NAME)
+            + " is found");
+      }
+    }
+
+    if (configuration.getString(ALERT_NAME) != null) {
+      List<AlertConfigDTO> duplicateAlerts = alertConfigDAO.findWhereNameLike(configuration.getString(ALERT_NAME));
+      for (AlertConfigDTO duplicateAlert : duplicateAlerts) {
+        if (duplicateAlert.getName().equals(configuration.getString(ALERT_NAME))) {
+          throw new IllegalArgumentException("Duplicate alert name " + configuration.getString(ALERT_NAME)
+              + " is found");
+        }
+      }
+    }
+
     // update datasetConfig
     DatasetConfigDTO datasetConfig = datasetConfigDAO.findByDataset(configuration.getString(COLLECTION_NAME));
     if (datasetConfig == null) {
       throw new NoSuchElementException("Cannot find collection: " + configuration.getString(COLLECTION_NAME));
     }
     TimeSpec timeSpec = ThirdEyeUtils.getTimeSpecFromDatasetConfig(datasetConfig);
     TimeGranularity dataGranularity = timeSpec.getDataGranularity();
+    TimeGranularity userAssignedDataGranularity = null;
```

Comment:
Followup on this question: User can input 1_MINUTES, 5_MINUTES, HOURS, DAYS as to specify granularity. Later we probably will also support 15_MINUTES. How can we parse this input? format as (X)_(MINUTES) from users?

Reply:
If we gives MINUTES, HOURS, or DAYS, the TimeGranularity can parse them directly to 1_MINUTES, 1_HOURS, or 1_DAYS.